docs/guides/primitive-input-output.ipynb

```diff
@@ -0,0 +1,461 @@
+{
```

Comment:
<p>Why is the format of the results dependent on the PUBs used and options? I would expect the format to be something fixed, or dependent on the primitive used. Are you referring to the metadata? I might be misunderstanding this sentence or reading too much into it.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Qiskit/documentation/pull/1621/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Reply:
<p>The format of the PUB results only promises to include <code>pub_result.data.evs</code> and <code>pub_result.data.stds</code>, but other pieces of data are included in the attributes of the <code>DataBin</code> if, for example ZNE is used (in this case it would include <code>pub_result.data.evs_extrapolated</code> and <code>pub_result.data.noise_factors</code>) . The metadata of the <code>PubResult</code> is also dependent entirely on the primitive and its options. This section is referring to both the PUB format and primitive options specified.</p>

Reply:
<p>@ElePT, if you consider the field names of the <code>pub_result.data</code> or  the <code>pub_result.shape</code> to be "format", then yes the format strongly depends on the input pub. For the sampler, the field names of the data come from classical registers. If you are talking about some other format, let's discuss. </p>